src/client-lib/utils/rendezvous-socket.js

```diff
@@ -0,0 +1,43 @@
+import EventEmitter from 'eventemitter3';
+
+class RendezvousSocket extends EventEmitter {
```

Comment:
If you want that, you could mix the behavior directly into `this` instead of attaching it to `this.events`.

One practical downside to mixing in directly (or using inheritance), is that you end up attaching into `RendezvousSocket` any private properties that `EventEmitter` (e.g. `_events`) may have. You also end up automatically attaching other functions (e.g. `setMaxListeners`) that you may want to use for your own logic instead of `EventEmitter`'s. For the common cases of `.emit()` or `.on()`, you could implement those on `RendezvousSocket` and pass-through to `this.events`.

W.r.t. inheritance specifically, this just feels like a _composition over inheritance_ situation but how much thats a personal preference or not, I'm not really sure.

We can discuss in person tomorrow and leave a note here. @mstemm gave you a +1 so feel free to merge this PR.